includes/preflight.inc

```diff
@@ -561,6 +561,18 @@ function drush_preflight_root() {
   // Load the config options from Drupal's /drush, ../drush, and sites/all/drush directories,
   // even prior to bootstrapping the root.
   drush_load_config('drupal');
+
+  // Search for commandfiles in the root locations
+  $discovery = annotationcommand_adapter_get_discovery();
+  $searchpath = [dirname($root) . '/drush', "$root/drush", "$root/sites/all/drush"];
```

Comment:
There are still > 100 calls to dirname in Drush. Perhaps we should look into switching those over in a separate PR.